### ISSUE TITLE
DllImportGenerator.UnitTests failure: The type or namespace name 'DllImportAttribute' could not be found

### ISSUE THREAD

**Issue by sbomer** [dotnet/runtime#64586]
Rolling build [failure](https://runfo.azurewebsites.net/view/build/?number=1582910), [logs](https://helixre8s23ayyeko0k025g8.blob.core.windows.net/dotnet-runtime-refs-heads-main-9e99d3723a284d068d/DllImportGenerator.Unit.Tests/1/console.f7c76d73.log?%3Fsv%3D2019-07-07%26se%3D2022-02-20T21%253A43%253A51Z%26sr%3Dc%26sp%3Drl%26sig%3DzI%252FrzjZLbVxWFW8NQ%252FhJdtXOXP7L7XSsiRWxG3WZpTg%253D) on net7.0-OSX-Release-x64-CoreCLR_release-OSX.1014.Amd64.Open. 
```
DllImportGenerator.UnitTests.Diagnostics.TargetFrameworkNotSupported_NoGeneratedDllImport_NoDiagnostic(targetFramework: Core) [FAIL]
      Assert.Empty() Failure
      Expected: <empty>
      Actual:   [(5,6): error CS0246: The type or namespace name 'DllImportAttribute' could not be found (are you missing a using directive or an assembly reference?), (5,6): error CS0246: The type or namespace name 'DllImport' could not be found (are you missing a using directive or an assembly reference?), (2,1): hidden CS8019: Unnecessary using directive.]
      Stack Trace:
        /_/src/libraries/System.Runtime.InteropServices/tests/DllImportGenerator.UnitTests/Diagnostics.cs(67,0): at DllImportGenerator.UnitTests.Diagnostics.TargetFrameworkNotSupported_NoGeneratedDllImport_NoDiagnostic(TestTargetFramework targetFramework)
        --- End of stack trace from previous location ---
```

This failure shows up in:
- DllImportGenerator.UnitTests.Diagnostics.TargetFrameworkNotSupported_NoDiagnostic
- DllImportGenerator.UnitTests.Diagnostics.TargetFrameworkNotSupported_NoGeneratedDllImport_NoDiagnostic
- DllImportGenerator.UnitTests.Compiles.ValidateSnippetsFallbackForwarder

The failure only shows up on OSX so far. This appears to be a new failure mode - I am seeing the first occurrence in the above pipeline run which has these new commits: https://github.com/dotnet/runtime/compare/bc6d349ecd72eec162f0532e6b6218fdbaa07ddb...4a19a4940d18fd71a4106f3bbdaf22a733790731. But none of those commits look related to me so it may just be nondeterministic. One of these tests previously had similar unusual failures on OSX: https://github.com/dotnet/runtime/issues/62467 (error CS0518: Predefined type 'System.Object' is not defined or imported, etc.).

/cc @AaronRobinsonMSFT

**Comment by AaronRobinsonMSFT** [comment]
@elinor-fung or @jkoritzinsky Did we change any failure cases where we would not find a type?

**Comment by jkoritzinsky** [comment]
No, we didn't.  I think this is another sign of the GC Hole that we've been seeing pop up here and there. I think this has the same root issue as #62467.
